web/modules/custom/user_role/user_role.module

```diff
@@ -0,0 +1,22 @@
+<?php
+
+use Drupal\Core\Form\FormStateInterface;
+/**
+ * Implements hook_form_alter().
+ */
+function user_role_form_alter(&$form, FormStateInterface $form_state, $form_id) {
```

Comment:
`array &$form`.

Reply:
In Drupal core, the hook is defined in this format. Then do I still need to define the datatype as you have mentioned?